crates/noirc_evaluator/src/ssa/acir_gen/acir_mem.rs

```diff
@@ -52,30 +198,54 @@ impl AcirMem {
         })
     }
 
+    // number of bits required to store the input
+    fn bits(mut t: usize) -> u32 {
+        let mut r = 0;
+        while t != 0 {
+            t >>= 1;
+            r += 1;
+        }
+        r
+    }
```

Comment:
Could be `usize::BITS - t.leading_zeroes()` or `t.ilog2()` but would need a check for `t == 0`

Reply:
Actually, should the number of bits for the value `0` really be `0` bits and not 1? Perhaps we need the check anyway:
```rs
if t == 0 { 0 } else { t.ilog2() + 1 }
```

Reply:
I searched for a built-in function for this but did not find it. This is a shame that I have to implement it myself, especially considering there are faster ways to do it. Even if it exists, it should be easy to find.

0 is not an issue here as it won't be called for 0. What I need is `ceil(log2(t))`, so ilog2 does not work.
I believe `usize::BITS - t.leading_zeroes()`should work but it looks more hacky, it does not work for 0 for instance.

Reply:
What we want is not `ceil(log2(t))` (nor `ceil(log2(t)) + 1`) here since:
```
ceil(log2(2)) = 1
ceil(log2(3)) = 2

ceil(log2(2)) + 1 = 2
ceil(log2(3)) + 1 = 3
```
While for the code you implemented:
```
bits(2) = 2
bits(3) = 2
```
What we want is indeed `ilog2(t) + 1` - ideally with the extra case for 0 to avoid panics.

Reply:
Ok yes, ilog2(t)+1 works